dz13/src/app.js

```diff
@@ -0,0 +1,126 @@
+function entriesArray(array) {
+  return array.flat(Infinity)
+}
+
+// function setKeysAndObjectInfo(user) {
+//   const entriesUser = Object.entries(user)
+//   const keyUser = []
+//   for (const key of entriesUser) {
+//     key.pop()
+//     keyUser.push(key)
+//   }
+//   console.log(keyUser.flat())
+// }
+function setKeysAndObjectInfo(user, type) {
+  if (type === 'key') {
+    const newUser = Object.keys(user)
+    return newUser
+  }
+  const newUser = Object.values(user)
+  return newUser
+}
+
+function logInCheck(cart) {
+  const newCart = Object.entries(cart)
+  for (const logCart of newCart) {
+    console.log(`${logCart[0]}: ${logCart[1]} шт.`)
+  }
+}
+
+function mergeSettings(defaultSettings, userPreferences) {
+  const resultSettings = { ...defaultSettings, ...userPreferences }
+  return resultSettings
+}
+
+function checkUserAge(user) {
+  try {
+    if (user.age < 18) {
+      throw new Error('Меньше 18 лет')
+    }
+    console.log('Больше 18 лет(')
+  } catch (error) {
+    if (error.message === 'Меньше 18 лет') {
+      console.log('Меньше 18 лет(')
+    }
+  }
+}
+
+function checkNumber(number) {
+  return new Promise((resolve, reject) => {
+    if (number > 10) {
+      resolve('Число больше 10')
+    } else {
+      reject('Число меньше или равно 10')
+    }
+  })
+}
+
+function delayedMessage(string, time) {
+  setTimeout(() => {
+    console.log(string)
+  }, time)
+}
+
+async function fetchProducts() {
+  try {
+    const data = await fetch('https://dummyjson.com/products?limit=10&select=title,price')
+    if (!data.ok) {
+      throw new Error('Не найден json')
+    }
+    const newData = await data.json()
+    console.log(newData.products[0])
+  } catch (error) {
+    console.log('Не удалось получить данные')
+  }
+}
```

Comment:
```suggestion
async function fetchProduct() {
  try {
    const response = await fetch('https://dummyjson.com/products?limit=10&select=title,price')

    if (!response.ok) {
      throw new Error('Ошибка при выполнении запроса')
    }

    const data = await response.json()

    console.log(data.products[0])
  } catch (error) {
    console.log('Не удалось получить продукт', error)
  }
}
```

результат `fetch` обычно называется ответ (`response`)

данные мы уже после и получаем используя `.json()`, который обрабатывает этот ответ доставая из него `body` и преобразуя в `json`

в `catch` можно показывать и ошибку `error` чтобы было легче понять, что за ошибка